core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -619,6 +619,9 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
         if (!unit.canEnterForeignTerrain && !tile.canCivPassThrough(unit.civInfo)) return false
 
         val firstUnit = tile.getFirstUnit()
+        // allow Ranged Units to capture Civilian
+        if (firstUnit != null && firstUnit.isCivilian() && unit.isRanged() )
```

Comment:
This is missing a check that the tile doesn't contain a military unit - don't count on getFirst to return the military unit if there is one

Reply:
ALSO
you don't check if the unit is off an ENEMY civ! You shouldn't be able to capture allied civilians with impunity

Reply:
> This is missing a check that the tile doesn't contain a military unit - don't count on getFirst to return the military unit if there is one

Why is that not guaranteed? Is this a multithreading thing? When I read getFirst() seemed it went in order.

Reply:
It is said that "programmers won't depends on what your library says it does, that will depend on observed results. Ordered results will be relief on to be ordered, and random ordering will be taken as a source of randomness".
Basically - if we want to be about to change the implementation of getFirst, we need to ensure that functions don't depend on a byproduct of its internal logic.